packages/governance/src/governor/extensions/governor_settings.cairo

```diff
@@ -0,0 +1,190 @@
+    use crate::governor::extensions::interface::IGovernorSettingsAdmin;
+    use openzeppelin_introspection::src5::SRC5Component;
+    use starknet::storage::{StoragePointerReadAccess, StoragePointerWriteAccess};
+
+    #[storage]
+    pub struct Storage {
+        pub Governor_voting_delay: u64,
+        pub Governor_voting_period: u64,
+        pub Governor_proposal_threshold: u256,
+    }
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    pub enum Event {
+        VotingDelayUpdated: VotingDelayUpdated,
+        VotingPeriodUpdated: VotingPeriodUpdated,
+        ProposalThresholdUpdated: ProposalThresholdUpdated
+    }
+
+    /// Emitted when `Governor_voting_delay` is updated.
+    #[derive(Drop, starknet::Event)]
+    pub struct VotingDelayUpdated {
+        pub old_voting_delay: u64,
+        pub new_voting_delay: u64
+    }
+
+    /// Emitted when `Governor_voting_period` is updated.
+    #[derive(Drop, starknet::Event)]
+    pub struct VotingPeriodUpdated {
+        pub old_voting_period: u64,
+        pub new_voting_period: u64
+    }
+
+    /// Emitted when `Governor_proposal_threshold` is updated.
+    #[derive(Drop, starknet::Event)]
+    pub struct ProposalThresholdUpdated {
+        pub old_proposal_threshold: u256,
+        pub new_proposal_threshold: u256
+    }
+
+    pub mod Errors {
+        pub const INVALID_VOTING_PERIOD: felt252 = 'Invalid voting period';
+    }
+
+    //
+    // Extensions
+    //
+
+    pub impl GovernorSettings<
+        TContractState,
+        +GovernorComponent::HasComponent<TContractState>,
+        +SRC5Component::HasComponent<TContractState>,
+        impl GovernorSettings: HasComponent<TContractState>,
+        +Drop<TContractState>
+    > of GovernorComponent::GovernorSettingsTrait<TContractState> {
+        /// See `GovernorComponent::GovernorSettingsTrait::voting_delay`.
+        fn voting_delay(self: @GovernorComponentState<TContractState>) -> u64 {
+            let contract = self.get_contract();
+            let this_component = GovernorSettings::get_component(contract);
+
+            this_component.Governor_voting_delay.read()
+        }
+
+        /// See `GovernorComponent::GovernorSettingsTrait::voting_period`.
+        fn voting_period(self: @GovernorComponentState<TContractState>) -> u64 {
+            let contract = self.get_contract();
+            let this_component = GovernorSettings::get_component(contract);
+
+            this_component.Governor_voting_period.read()
+        }
+
+        /// See `GovernorComponent::GovernorSettingsTrait::proposal_threshold`.
+        fn proposal_threshold(self: @GovernorComponentState<TContractState>) -> u256 {
+            let contract = self.get_contract();
+            let this_component = GovernorSettings::get_component(contract);
+
+            this_component.Governor_proposal_threshold.read()
+        }
+    }
+
+    //
+    // External
+    //
+
+    #[embeddable_as(SetSettingsImpl)]
+    impl SetSettings<
```

Comment:
We could name the trait more in line with the name of the trait being implemented (`IGovernorSettingsAdmin`)

Reply:
Good catch, I miss that as part of the name update.